sdcm/cluster_k8s/__init__.py

```diff
@@ -438,11 +438,13 @@ def wait_for_init(self):
     def wait_for_pods_readiness(self, count: Optional[int] = None):
         if count is None:
             count = len(self.nodes)
-        for _ in range(count):
-            time.sleep(self.pod_readiness_delay)
-            self.k8s_cluster.kubectl(f"wait --timeout={self.pod_readiness_timeout}m --all --for=condition=Ready pod",
-                                     namespace=self.namespace,
-                                     timeout=self.pod_readiness_timeout*60+10)
+        time.sleep(self.pod_readiness_delay)
+        result = self.k8s_cluster.kubectl(
+            f"wait --timeout={self.pod_readiness_timeout}m --all --for=condition=Ready pod",
+            namespace=self.namespace,
+            timeout=self.pod_readiness_timeout * 60 + 10)
+        if result.stdout.count('condition met') != count:
```

Comment:
There is another problem: you need to handle situation when we grow cluster. In that case count will be equal to number of added nodes, and count of "condition met" to total nodes.

Reply:
Nice finding, fixed.